src/generators/Silk.NET.SilkTouch.Symbols/StructSymbol.cs

```diff
@@ -9,11 +9,31 @@ namespace Silk.NET.SilkTouch.Symbols;
 /// A <see cref="TypeSymbol"/> representing a <c>struct</c>.
 /// </summary>
 /// <param name="Identifier">The Identifier of this struct</param>
-/// <param name="Members">The Members of this struct</param>
+/// <param name="Layout">The layout of this struct</param>
 /// <remarks>
 /// In this context, a Struct means a type that represents the layout of a continuous block of memory.
 /// </remarks>
 // /// Each meaningful place in this memory called a field (see <see cref="FieldSymbol"/>) is accessible via this type.
 // /// Fields are allowed to overlap.
 // /// Additionally it may contain one or multiple <see cref="MethodSymbol"/> that are called with an instance of this type as their first argument.
-public sealed record StructSymbol(IdentifierSymbol Identifier, ImmutableArray<MemberSymbol> Members) : TypeSymbol(Identifier);
+public sealed record StructSymbol(IdentifierSymbol Identifier, StructLayout Layout) : TypeSymbol(Identifier);
+
+/// <summary>
+/// A <see cref="StructSymbol"/> representing the layout of a <see cref="StructSymbol"/>
+/// </summary>
+/// <param name="Entries">The entries of this layout</param>
+public sealed record StructLayout(ImmutableArray<LayoutEntry> Entries)
+{
```

Comment:
Did we want to support auto layout in cases where we don't want to guarantee the memory layout to our users?